models/intermediate/int_apple_store__date_spine.sql

```diff
@@ -0,0 +1,49 @@
+{{ config(materialized='table') }}
+
+-- depends_on: {{ ref('stg_apple_store__app_store_discovery_and_engagement_daily') }}
+-- depends_on: {{ ref('stg_apple_store__app_crash_daily') }}
+-- depends_on: {{ ref('stg_apple_store__app_store_download_daily') }}
+-- depends_on: {{ ref('stg_apple_store__app_store_installation_and_deletion_daily') }}
+-- depends_on: {{ ref('stg_apple_store__app_session_daily') }}
+with spine as (
+
+    {% if execute and flags.WHICH in ('run', 'build') %}
+
+{% set first_date_query %}
+
+    select min(date_day) as min_date_day
+    from (
+        select date_day from {{ ref('stg_apple_store__app_store_discovery_and_engagement_daily') }}
```

Comment:
Is there a reason we need to select all records from these? Why not just the following to save on performance.
```suggestion
        select min(date_day) from {{ ref('stg_apple_store__app_store_discovery_and_engagement_daily') }}
```